conditional_camera/conditional_cam.go

```diff
@@ -134,47 +133,37 @@ func (cc *conditionalCamera) Images(ctx context.Context) ([]camera.NamedImage, r
 	return nil, meta, data.ErrNoCaptureToStore
 }
 
-func (cc *conditionalCamera) Stream(ctx context.Context, errHandlers ...gostream.ErrorHandler) (gostream.VideoStream, error) {
-	camStream, err := cc.cam.Stream(ctx, errHandlers...)
+func (cc *conditionalCamera) Image(ctx context.Context, mimeType string, extra map[string]interface{}) ([]byte, camera.ImageMetadata, error) {
+
+	bytes, meta, err := cc.cam.Image(ctx, mimeType, extra)
 	if err != nil {
-		return nil, err
+		return nil, meta, err
 	}
-
-	return conditionalStream{camStream, cc}, nil
-}
-
-type conditionalStream struct {
-	cameraStream gostream.VideoStream
-	cc           *conditionalCamera
-}
-
-func (cs conditionalStream) Next(ctx context.Context) (image.Image, func(), error) {
-	extra, ok := camera.FromContext(ctx)
-	if !ok || extra[data.FromDMString] != true {
-		// If not data management collector, return underlying stream contents without filtering.
-		return cs.cameraStream.Next(ctx)
+	ex, ok := ctx.Value(0).(map[string]interface{})
+	if !ok || ex[data.FromDMString] != true {
+		return bytes, meta, err
 	}
 
-	img, release, err := cs.cameraStream.Next(ctx)
+	img, err := rimage.DecodeImage(ctx, bytes, mimeType)
```

Comment:
same here, do not use the mimeType you used in the request. `Image` may return a different mimeType. Make a unit test for this too.